nequip/nn/_graph_mixin.py

```diff
@@ -295,7 +295,7 @@ def insert(
             assert AtomicDataDict._irreps_compatible(
                 module_list[idx - 1].irreps_out, module.irreps_in
             )
-        if len(module_list) > idx:
+        if len(module_list) - 1 > idx:
```

Comment:
why?

Reply:
the next line is checking the `idx+1` layer.

Reply:
:+1: